content/gotraining.slide

```diff
@@ -613,7 +613,7 @@ The calling code should handle errors by testing whether the error equals nil:
 
 Panic should be used when something went unexpectedly wrong.
 For instance, initialise a regular expression with [[https://golang.org/pkg/regexp/#MustCompile][regexp.MustCompile]] if the pattern is a string constant.
-`MustCompile` will `panic` during `go` `build` if the pattern is invalid.
+`MustCompile` will `panic` at initialisation.
```

Comment:
i was surprised about this initially (compile time regex checks) but i somehow convinced myself it did it (without knowing how). but i did run some tests that i can no longer find, and i swear i saw a compile time failure. maybe i did it in the playground and confused compile time with run time, but i can no longer reproduce what i thought i saw.

sorry i should have not left this hanging, and actually figured out how it would have worked.